### DISCUSSION TITLE
entry point  and main controller

### DISCUSSION THREAD

**Discussion by farhad-sh** [joomengine/Joomla-Component-Builder#659] [Extention Issues]
can i insert my custom code in site/administrator entry point or main controller by jcb?
how?

**Comment by Llewellynvdm** [comment]
JCB is very smart and very advanced tool, you can do that which you are asking an so much more. But you will need to do the work of learning JCB by watching it tutorials. I can't write out to you what I have already spend hours to explain in tutorials. So we converted all the tutorials to text, so you can search it like this:
https://github.com/vdm-io/Joomla-Component-Builder/search?q=import-custom-code&type=wikis

Here are some the tutorial you may want to look at first:
https://www.youtube.com/watch?v=DFMfIl-VkSk&list=PLQRGFI8XZ_wtGvPQZWBfDzzlERLQgpMRE&index=31
https://www.youtube.com/watch?v=KiAtJawZ3oo&list=PLQRGFI8XZ_wtGvPQZWBfDzzlERLQgpMRE&index=32
https://www.youtube.com/watch?v=FEJ8WEbcTys&list=PLQRGFI8XZ_wtGvPQZWBfDzzlERLQgpMRE&index=87
https://www.youtube.com/watch?v=TjoH5rj4UgM&list=PLQRGFI8XZ_wsV6PDFXd21B7O0X70JlNd3&index=7

This should show you that you can change any code JCB produce, but I would give you this tutorial to advice the best practice:
https://www.youtube.com/watch?v=QGMURJI_Mds&list=PLQRGFI8XZ_wuzAQELTUUIDdqKFmLP9lSl&index=10
